src/components/SingleArticle/Comments.jsx

```diff
@@ -0,0 +1,89 @@
+import { getComments, postComment } from "../../api";
+import { useEffect } from "react";
+import { useState } from "react";
+import { Button, Group, Stack, Textarea } from "@mantine/core";
+import CommentCard from "./CommentCard";
+import { useForm } from "@mantine/form";
+import { notifications } from "@mantine/notifications";
+
+const Comments = ({ articleId }) => {
+  const [comments, setComments] = useState([]);
+  const [sendingComment, setSendingComment] = useState(false);
+
+  useEffect(() => {
+    getComments(articleId).then((commentResponse) => {
+      setComments(commentResponse);
+    });
+  }, [articleId]);
+  const form = useForm({
+    mode: "uncontrolled",
+    initialValues: {
+      body: "",
+    },
+
+    validate: {
+      body: (value) =>
+        value.length >= 4
+          ? null
+          : "Your comment must have at least 4 characters!",
+    },
+  });
+
+  const handleCommentSubmit = (values) => {
+    setSendingComment(true);
+
+    const username = "grumpy19";
+    const body = values.body;
+
+    postComment(articleId, username, body).then((comment) => {
+      setComments((prevComments) => {
+        form.reset();
+        setSendingComment(false);
+
+        return [comment, ...prevComments];
+      });
+      notifications.show({
+        title: "Comment posted!",
+        message: `Thank you for sharing your thoughts, ${username}!`,
+      });
+    });
+  };
```

Comment:
I think you should extract the logic/html handling the post comment into its own component, to make this component less busy.